src/librustc_typeck/structured_errors.rs

```diff
@@ -147,3 +147,41 @@ https://doc.rust-lang.org/reference/expressions/operator-expr.html#type-cast-exp
         err
     }
 }
+
+pub struct CEnumImplDropCastError<'tcx> {
+    sess: &'tcx Session,
+    span: Span,
+    expr_ty: Ty<'tcx>,
+    cast_ty: String,
+}
+impl<'tcx> CEnumImplDropCastError<'tcx> {
+    pub fn new(
+        sess: &'tcx Session,
+        span: Span,
+        expr_ty: Ty<'tcx>,
+        cast_ty: String,
+    ) -> CEnumImplDropCastError<'tcx> {
+        CEnumImplDropCastError { sess, span, expr_ty, cast_ty }
+    }
+}
+
+impl<'tcx> StructuredDiagnostic<'tcx> for CEnumImplDropCastError<'tcx> {
+    fn session(&self) -> &Session {
+        self.sess
+    }
+
+    fn code(&self) -> DiagnosticId {
+        rustc_errors::error_code!(E0649)
+    }
+
+    fn common(&self) -> DiagnosticBuilder<'tcx> {
+        self.sess.struct_span_fatal_with_code(
```

Comment:
This should not use `*_fatal_*`, as that prevents other errors from being emitted.

Reply:
Thanks for pointing that out.